### PR TITLE
[MOD:BREAK] Update matplotlib to fix deprecated functions

### PR DESCRIPTION
**Pull request by b-batton-sw**

### CODE REVIEW

File: src/statworx_theme/utils.py [hunk 1]
```diff
@@ -116,8 +117,10 @@ def get_stwx_cmaps(as_hex=True) -> Dict[str, Any]:
     cmap_names = [cmap for cmap in plt.colormaps() if cmap.startswith("stwx:")]
     cmaps = [plt.get_cmap(cmap) for cmap in cmap_names]
     if as_hex:
+        from matplotlib import colors
```
Comment (rbange) [inline src/statworx_theme/utils.py:120]: Why is matplotlib imported within the function? Its a hard dependency anyway.

**Review by rbange — CHANGES_REQUESTED** [review]
What was the reason this package is so python version constrained anyhow?

File: src/statworx_theme/utils.py (hunk 1 above)
Comment (b-batton-sw) [reply src/statworx_theme/utils.py:120]: Good catch, I was wondering why mpl.colors.to_hex was not working and imported colors to test it, but forgot to remove

**Review by rbange — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -14,7 +14,7 @@ repository = "https://github.com/STATWORX/statworx-theme"
 homepage = "https://statworx-theme.readthedocs.io/en/latest"
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
+python = "~3.11"
```
Comment (DominiqueLade) [inline pyproject.toml:17]: Why do we limit it to Pyhotn 3.11 only?

**Review by DominiqueLade — CHANGES_REQUESTED** [review]

**Review by rbange — COMMENTED** [review]
I guess we need to remove this dependency from the project template then... This limitation would otherwise hard limit all projects to python 3.11

File: pyproject.toml (hunk 2 above)
Comment (b-batton-sw) [reply pyproject.toml:17]: I guess I was moving to quickly here.

With the prior dependency an update of matplotlib to the required version was not possible, ayn ideas why?

Comment (b-batton-sw) [reply pyproject.toml:17]: The version bump action failed anyhow, so deployment isn't done yet

**Pull request merged**